nixos/modules/services/security/authentik.nix

```diff
@@ -0,0 +1,207 @@
+
+        Authentik requires at least a secret key:
+        ```
+        AUTHENTIK_SECRET_KEY=<random key>
+        ```
+      '';
+    };
+    environment = mkOption {
+      type = types.submodule { freeformType = types.attrsOf types.str; };
+      default = { };
+      example = {
+        AUTHENTIK_LOG_LEVEL = "debug";
+      };
+      description = ''
+        Extra configuration environment variables. Refer to the [documentation](https://docs.goauthentik.io/docs/installation/configuration) for options.
+      '';
+    };
+
+    host = lib.mkOption {
+      type = types.str;
+      default = "localhost";
+      description = "The host that authentik will listen on.";
+    };
+    port = mkOption {
+      type = types.port;
+      default = 9000;
+      description = "The port that immich will listen on.";
+    };
+    user = mkOption {
+      type = types.str;
+      default = "authentik";
+      description = "The user immich should run as.";
+    };
+    group = mkOption {
+      type = types.str;
+      default = "authentik";
+      description = "The group immich should run as.";
+    };
+    database = {
+      enable =
+        mkEnableOption "the postgresql database for use with authentik. See {option}`services.postgresql`"
+        // {
+          default = true;
+        };
+      createDB = mkEnableOption "the automatic creation of the database for authentik." // {
+        default = true;
+      };
+      name = mkOption {
+        type = types.str;
+        default = "authentik";
+        description = "The name of the authentik database.";
+      };
+      host = mkOption {
+        type = types.str;
+        default = "/run/postgresql";
+        example = "127.0.0.1";
+        description = "Hostname or address of the postgresql server. If an absolute path is given here, it will be interpreted as a unix socket path.";
+      };
+      port = mkOption {
+        type = types.port;
+        default = 5432;
+        description = "The port of the postgresql server";
+      };
+      user = mkOption {
+        type = types.str;
+        default = "authentik";
+        description = "The database user for authentik.";
+      };
+    };
+    redis = {
+      enable = mkEnableOption "a redis cache for use with authentik" // {
+        default = true;
+      };
+      host = mkOption {
+        type = types.str;
+        default = config.services.redis.servers.authentik.unixSocket;
+        defaultText = lib.literalExpression "config.services.redis.servers.authentik.unixSocket";
+        description = "The host that redis will listen on.";
+      };
+      port = mkOption {
+        type = types.port;
+        default = 0;
+        description = "The port that redis will listen on. Set to zero to disable TCP.";
+      };
+    };
```

Comment:
Should this be like the DB, with a specific option for local provisioning?

Reply:
What do you mean?